### PR TITLE
Fixes and updates 

### PR DESCRIPTION
**Pull request by KateMoreva**
- bdv v 10.2.2
https://github.com/mobie/mobie-io/issues/94

- doesObjectExist removed in N5S3OMEZarrReader
  https://github.com/mobie/mobie-io/issues/91

- tried to fix bug with bdv.ome.zarr

### CODE REVIEW

File: src/main/java/org/embl/mobie/io/SpimDataOpener.java [hunk 1]
```diff
@@ -156,18 +157,40 @@ private SpimData openOpenOrganelleS3(String path) throws SpimDataException {
...
             }
-        } catch (IOException | JDOMException e) {
-            throw new SpimDataException(ERROR_WHILE_TRYING_TO_READ_SPIM_DATA + e.getMessage());
+            SpimData spim = new SpimData(null, Cast.unchecked(imageLoader.getSequenceDescription()), imageLoader.getViewRegistrations());
+            SpimData sp1;
+            try {
+                InputStream st = FileAndUrlUtils.getInputStream(path);
+                sp1 = (new CustomXmlIoSpimData()).loadFromStream(st, path);
+            } catch (SpimDataException var3) {
+                System.out.println(path);
```
Comment (tischi) [inline src/main/java/org/embl/mobie/io/SpimDataOpener.java:183]: @KateMoreva 
Could you replace `var3` by a more meaningful name?
In this case `spimDataException`.
Are you using IntelliJ? This usually recommend good variable names.
In general, names like `sp1` and `var4` make the code very hard to read, please try using variable names that explain what the content of the variable is.

File: src/main/java/org/embl/mobie/io/util/CustomXmlIoSpimData.java [hunk 2]
```diff
@@ -41,28 +41,27 @@
...
             doc = sax.build(in);
-        } catch (final Exception e) {
-            throw new SpimDataIOException(e);
+        } catch (Exception var6) {
+            throw new SpimDataIOException(var6);
         }
         final Element root = doc.getRootElement();
 
-        if (root.getName() != SPIMDATA_TAG)
+        if (!Objects.equals(root.getName(), SPIMDATA_TAG))
```
Comment (tischi) [inline src/main/java/org/embl/mobie/io/util/CustomXmlIoSpimData.java:64]: Is that a String comparison?
Why don't you use: `root.getName().equals(SPIMDATA_TAG)`?

**Review by tischi — COMMENTED** [review]
I had a few comments.
Otherwise looks good. 
Thanks!

File: src/main/java/org/embl/mobie/io/SpimDataOpener.java (hunk 1 above)
Comment (KateMoreva) [reply src/main/java/org/embl/mobie/io/SpimDataOpener.java]: This code does not fix all problems. I'll rewrite it in the next PR.

File: src/main/java/org/embl/mobie/io/util/CustomXmlIoSpimData.java (hunk 2 above)
Comment (KateMoreva) [reply src/main/java/org/embl/mobie/io/util/CustomXmlIoSpimData.java]: You are right, root.getName().equals(SPIMDATA_TAG) is better

File: src/main/java/org/embl/mobie/io/util/CustomXmlIoSpimData.java [hunk 3]
```diff
@@ -41,28 +41,27 @@
...
-        final SAXBuilder sax = new SAXBuilder();
+        SAXBuilder sax = new SAXBuilder();
+
         Document doc;
         try {
             doc = sax.build(in);
-        } catch (final Exception e) {
-            throw new SpimDataIOException(e);
+        } catch (Exception var6) {
+            throw new SpimDataIOException(var6);
```
[the remark this answers is not in the document]
Comment (KateMoreva) [reply src/main/java/org/embl/mobie/io/util/CustomXmlIoSpimData.java]: fixed

File: src/main/java/org/embl/mobie/io/SpimDataOpener.java [hunk 4]
```diff
@@ -156,18 +159,39 @@ private SpimData openOpenOrganelleS3(String path) throws SpimDataException {
...
+            } catch (SpimDataException exception) {
+                log.debug("Failed to load stream from {}", path, exception);
+                return null;
+            }
+            sp1.setBasePath(null);
+            sp1.getSequenceDescription().setImgLoader(spim.getSequenceDescription().getImgLoader());
+            sp1.getSequenceDescription().getAllChannels().putAll(spim.getSequenceDescription().getAllChannels());
+            return sp1;
+        } catch (JDOMException | IOException var13) {
+            var13.printStackTrace();
```
Comment (tischi) [inline src/main/java/org/embl/mobie/io/SpimDataOpener.java:193]: replace `var13`

File: src/main/java/org/embl/mobie/io/SpimDataOpener.java [hunk 5]
```diff
@@ -156,18 +159,39 @@ private SpimData openOpenOrganelleS3(String path) throws SpimDataException {
...
+                imageLoader = new N5S3OMEZarrImageLoader(imgLoaderElem.getChild("ServiceEndpoint").getText(), imgLoaderElem.getChild("SigningRegion").getText(), bucket, object, ".", queue);
             } else {
-                throw new SpimDataException("Error while trying to read spimData. SpimData is null");
+               imageLoader = new N5S3OMEZarrImageLoader(imgLoaderElem.getChild("ServiceEndpoint").getText(), imgLoaderElem.getChild("SigningRegion").getText(), bucket, object, ".");
             }
-        } catch (IOException | JDOMException e) {
-            throw new SpimDataException(ERROR_WHILE_TRYING_TO_READ_SPIM_DATA + e.getMessage());
+            SpimData spim = new SpimData(null, Cast.unchecked(imageLoader.getSequenceDescription()), imageLoader.getViewRegistrations());
+            SpimData sp1;
+            try {
```
Comment (tischi) [inline src/main/java/org/embl/mobie/io/SpimDataOpener.java:181]: rename `sp1`

File: src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java [hunk 6]
```diff
@@ -266,18 +267,15 @@ public DataBlock<?> readBlock(
...
+        try {
             InputStream in = this.readS3Object(objectPath);
             Throwable var4 = null;
 
-            HashMap<String, JsonElement> var5;
+            HashMap<String, JsonElement> hashMap;
             try {
-                var5 = GsonAttributesParser.readAttributes(new InputStreamReader(in), this.gson);
+                hashMap = GsonAttributesParser.readAttributes(new InputStreamReader(in), this.gson);
             } catch (Throwable var14) {
```
Comment (tischi) [inline src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java:279]: rename `var14`

File: src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java [hunk 7]
```diff
@@ -266,18 +267,15 @@ public DataBlock<?> readBlock(
...
-     * @throws IOException
      */
-    public HashMap<String, JsonElement> readJson(String objectPath) throws IOException {
-        if (!this.s3.doesObjectExist(this.bucketName, objectPath)) {
-            return null;
-        } else {
+    public HashMap<String, JsonElement> readJson(String objectPath) {
+        try {
             InputStream in = this.readS3Object(objectPath);
             Throwable var4 = null;
```
Comment (tischi) [inline src/main/java/org/embl/mobie/io/ome/zarr/readers/N5S3OmeZarrReader.java:274]: rename `var4`

**Review by tischi — COMMENTED** [review]
I found sub-optimal variable names at some other places.

**Comment by KateMoreva** [comment]
> I found sub-optimal variable names at some other places.

Fixed.

**Pull request merged**